spot_driver/spot_driver/spot_ros2.py

```diff
@@ -1466,245 +1470,216 @@ def handle_max_vel(self, request: SetVelocity.Request, response: SetVelocity.Res
+
+    def _process_synchronized_gripper_command_feedback(
+        self, feedback: GripperCommandFeedback
+    ) -> Optional[GoalResponse]:
+        if (
+            feedback.status.value == feedback.status.STATUS_COMMAND_OVERRIDDEN
+            or feedback.status.value == feedback.status.STATUS_COMMAND_TIMED_OUT
+            or feedback.status.value == feedback.status.STATUS_ROBOT_FROZEN
+            or feedback.status.value == feedback.status.STATUS_INCOMPATIBLE_HARDWARE
+        ):
+            return GoalResponse.FAILED
+
+        if feedback.command.command_choice == feedback.command.COMMAND_CLAW_GRIPPER_FEEDBACK_SET:
+            if (
+                feedback.command.claw_gripper_feedback.status.value
+                == feedback.command.claw_gripper_feedback.status.STATUS_IN_PROGRESS
             ):
-                if (
-                    full_body_feedback.feedback.payload_estimation_feedback.status.value
-                    == full_body_feedback.feedback.payload_estimation_feedback.status.STATUS_COMPLETED
-                ):
-                    return GoalResponse.SUCCESS
-                if (
-                    full_body_feedback.feedback.payload_estimation_feedback.status.value
-                    == full_body_feedback.feedback.payload_estimation_feedback.status.STATUS_SMALL_MASS
-                ):
-                    return GoalResponse.SUCCESS
-                if (
-                    full_body_feedback.feedback.payload_estimation_feedback.status.value
-                    == full_body_feedback.feedback.payload_estimation_feedback.status.STATUS_ERROR
-                ):
-                    return GoalResponse.FAILED
                 return GoalResponse.IN_PROGRESS
-            elif (
-                full_body_feedback.feedback.feedback_choice
-                == full_body_feedback.feedback.FEEDBACK_CONSTRAINED_MANIPULATION_FEEDBACK_SET
+            if (
+                feedback.command.claw_gripper_feedback.status.value
+                == feedback.command.claw_gripper_feedback.status.STATUS_UNKNOWN
             ):
-                if (
-                    full_body_feedback.feedback.constrained_manipulation_feedback.status.value
-                    == full_body_feedback.feedback.constrained_manipulation_feedback.status.STATUS_RUNNING
-                ):
-                    return GoalResponse.IN_PROGRESS
-                return GoalResponse.FAILED
-            else:
+                self.get_logger().error("ERROR: claw grippper status unknown")
                 return GoalResponse.IN_PROGRESS
+            # else: STATUS_AT_GOAL or STATUS_APPLYING_FORCE
+        else:
+            self.get_logger().error("ERROR: unknown gripper command type")
+            return GoalResponse.IN_PROGRESS
+        return None
 
-        elif feedback.command.command_choice == feedback.command.COMMAND_SYNCHRONIZED_FEEDBACK_SET:
+    def _robot_command_goal_complete(self, feedback: RobotCommandFeedback) -> GoalResponse:
```

Comment:
Oh right now I remember multiple feedback commands can be set so that's why you can't just return from one sub-feedback call.  But I think maybe having the `_process` return SUCCESS instead of None on SUCCESS would still be a lot clearer...